### ISSUE TITLE
Lucene cannot return more than 500 results no matter what QueryOptions "take" value is used

### ISSUE THREAD

**Issue by argentini** [Shazwazza/Examine#241]
### Which *exact* Umbraco version are you using? For example: 8.13.1 - don't just write v8

v9 rc002, Examine 2.0.0 beta 154 and 156

### Bug summary

Lucene cannot return more than 500 results no matter what QueryOptions "take" value is used.

### Steps to reproduce

An example of this is below. Using int.MaxValue, or some other value above 500, only returns up to 500 results.

```
if (ExamineManager.TryGetIndex("ExternalIndex", out var index) || !(index is { } umbIndex))
{
    throw new InvalidOperationException($"No index found by name ExternalIndex or is not of type {typeof(IUmbracoIndex)}");
}

var searcher = (BaseLuceneSearcher)umbIndex.Searcher;

var luceneOptions = new LuceneSearchOptions
{
    AllowLeadingWildcard = true
};

var qry = searcher.CreateQuery("content", BooleanOperation.And, searcher.LuceneAnalyzer, luceneOptions).NativeQuery(luceneQuery);
var results = qry.Execute(new QueryOptions(0, int.MaxValue));
```

### Expected result / actual result

As with Umbraco v8, it should return more than 500 results.

**Comment by Shazwazza** [comment]
Thanks for the report. This is now fixed in rev 2a51f1c70ce502b491d8e84e106d84f30f846cae

**Comment by Shazwazza** [comment]
Yes, it's this PR https://github.com/umbraco/Umbraco-CMS/pull/10962. And no, it cannot run in RC2, there are breaking changes.
